src/adhara/interop-service/corda-decoder/src/main/java/io/adhara/poc/ledger/SecureHash.java

```diff
@@ -0,0 +1,127 @@
+package io.adhara.poc.ledger;
+
+import io.adhara.poc.utils.Utils;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.validation.constraints.NotNull;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.stream.IntStream;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
```

Comment:
Why do we want NoArgs constructors for these types? And is that why otherwise `final`-looking private fields aren't declared as `final`?